pydicom/data/base.py

```diff
@@ -0,0 +1,51 @@
+"""pydicom data manager"""
+#
+# Copyright (c) 2008-2012 Darcy Mason
+# Copyright (c) 2017 pydicom AUTHORS
+# This file is part of pydicom, released under a modified MIT license.
+#    See the file LICENSE included with this distribution, also
+#    available at https://github.com/pydicom/pydicom
+#
+
+import fnmatch
+import os
+
+from .utils import get_files
+
+
+def get_datadir():
+    return os.path.abspath(os.path.dirname(__file__))
+
+
+def get_testdata_base():
+    return os.path.join(get_datadir(), 'test_files')
+
+
+def get_charset_base():
+    return os.path.join(get_datadir(), 'charset_files')
+
+
+'''
+
+Data get functions.
+
+For each of the below, a complete list of files is returned.
+Note the distinction between "get" (returning a list) and
+what might be interpreted as reading in the files (load).
+These functions serve to only provide the paths.
+
+Optionally, the user can specify a pattern to filter by.
+
+'''
+
+
+def get_charset_files(pattern=None):
```

Comment:
Glob with recursive is only supported for python 3.5+ https://docs.python.org/3/library/glob.html

>> Changed in version 3.5: Support for recursive globs using “**”.

I think it's pretty standard to use something that looks like a function to walk over the files. Do you have issue with that thing sharing a common function name and prefer repeating the same code over per data fetcher? Take a look at [scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/datasets/base.py#L71) equivalent, "load_files" It has the same spirit of the get_files function, but is far more robust. 

I tend to prefer developing in favor of code being maximally extendable for future use cases, but I don't think it's a good idea to use a function that isn't supported, and I don't like copy pasting the same code twice that could better do in a function. I'm indifferent otherwise and would be happy to following your suggestion, if it seems reasonable.